frontend/model/contracts/group.js

```diff
@@ -1044,6 +1044,8 @@ sbp('chelonia/defineContract', {
         const isUpdatingNonMonetary = Object.keys(data).some(
           key => ['nonMonetaryAdd', 'nonMonetaryRemove', 'nonMonetaryEdit', 'nonMonetaryReplace'].includes(key)
         )
+
+        const isChagingIncomeDetailsType = data.incomeDetailsType && groupProfile.incomeDetailsType !== data.incomeDetailsType
```

Comment:
The first check for `data.incomeDetailsType` is done again down below on line 1087. Since `isChagingIncomeDetailsType` is not used anywhere else except inside of that `if` block, could you please move this inside of the `if` block and redefine it as:

```js
const isChagingIncomeDetailsType = groupProfile.incomeDetailsType !== data.incomeDetailsType
```